cg/meta/workflow/mutant/quality_controller/utils.py

```diff
@@ -0,0 +1,108 @@
+from cg.apps.lims.api import LimsAPI
+from cg.constants.constants import MutantQC
+from cg.meta.workflow.mutant.constants import QUALITY_REPORT_FILE_NAME
+from cg.meta.workflow.mutant.metadata_parser.metadata_parser import MetadataParser
+from cg.meta.workflow.mutant.metadata_parser.models import SampleMetadata, SamplesMetadataMetrics
+from cg.meta.workflow.mutant.metrics_parser.metrics_parser import MetricsParser
+from cg.meta.workflow.mutant.metrics_parser.models import SamplesResultsMetrics
+from cg.meta.workflow.mutant.quality_controller.models import (
+    QualityMetrics,
+    SampleQualityResult,
+)
+from cg.store.models import Case, Sample
+
+from pathlib import Path
+
+from cg.store.store import Store
+
+
+def has_valid_total_reads(sample_metadata: SampleMetadata) -> bool:
+    if sample_metadata.is_external_negative_control:
+        if is_valid_total_reads_for_external_negative_control(reads=sample_metadata.reads):
+            return True
+        else:
+            return False
+
+    if sample_metadata.is_internal_negative_control:
+        return is_valid_total_reads_for_internal_negative_control(reads=sample_metadata.reads)
+
+    return is_valid_total_reads(
+        reads=sample_metadata.reads,
+        target_reads=sample_metadata.target_reads,
+        threshold_percentage=sample_metadata.percent_reads_guaranteed,
+    )
+
+
+def is_valid_total_reads(reads: int, target_reads: int, threshold_percentage: int) -> bool:
```

Comment:
I think this name clashes a bit with `has_valid_total_reads`.

Reply:
Changed!